rules/DowngradePhp70/Rector/FuncCall/DowngradeParenthesizedValueCallToCallUserFuncRector.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DowngradePhp70\Rector\FuncCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Name;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://stackoverflow.com/a/1596250/1348344
+ *
+ * @see \Rector\Tests\DowngradePhp70\Rector\FuncCall\DowngradeParenthesizedValueCallToCallUserFuncRector\DowngradeParenthesizedValueCallToCallUserFuncRectorTest
+ */
+final class DowngradeParenthesizedValueCallToCallUserFuncRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Downgrade calling a callable property disambiguated from calling a method using parentheses to call_user_func.',
+            [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+final class Foo
+{
+    /** @var callable handler for sending emails */
+    public $mailer;
+}
+
+$foo = new Foo;
+($foo->mailer)($args);
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+final class Foo
+{
+    /** @var callable handler for sending emails */
+    public $mailer;
+}
+
+$foo = new Foo;
+call_user_func($foo->mailer, $args);
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [FuncCall::class];
+    }
+
+    /**
+     * @param FuncCall $node
+     */
+    public function refactor(Node $node): ?FuncCall
+    {
+        if (! $node->name instanceof PropertyFetch) {
+            return null;
+        }
+
+        $args = array_merge([new Arg($node->name)], $node->args);
+
+        return new FuncCall(new Name('call_user_func'), $args);
```

Comment:
You can use NodeFactory service

https://github.com/rectorphp/rector-src/blob/b76243b41682edd072da3ca149461283c779415b/src/PhpParser/Node/NodeFactory.php#L417

Reply:
I guess we need a rector for that :rofl: since I copied it from another rector :rofl:

Reply:
Not sure if we need it though, since the arguments are already `Arg[]`. Also what is the difference between `NodeFactory::createFuncCall` and `BuilderFactory::funcCall`?

Reply:
both are helper shortcut to create FuncCall, `BuilderFactory::funcCall` probably too heavy for a certainly passed string name and arg.